assets/js/components/HostsList.test.jsx

```diff
@@ -99,6 +99,97 @@ describe('HostsLists component', () => {
     });
   });
 
+  describe('deregistration', () => {
+    it('should show the clean up button when the host is deregisterable', () => {
+      const host1 = hostFactory.build({ deregisterable: true });
+      const host2 = hostFactory.build({ deregisterable: false });
+      const state = {
+        ...defaultInitialState,
+        hostsList: {
+          hosts: [].concat(host1, host2),
+        },
+      };
+
+      const [StatefulHostsList] = withState(<HostsList />, state);
+
+      renderWithRouter(StatefulHostsList);
+      const table = screen.getByRole('table');
+      const cleanUpCell1 = table.querySelector(
+        'tr:nth-child(1) > td:nth-child(9)'
+      );
+      const cleanUpCell2 = table.querySelector(
+        'tr:nth-child(2) > td:nth-child(9)'
+      );
+      expect(cleanUpCell1).toHaveTextContent('Clean up');
+      expect(cleanUpCell2).not.toHaveTextContent('Clean up');
+    });
+
+    it('should show the host in deregistering state', () => {
+      const host = hostFactory.build({
+        deregisterable: true,
+        deregistering: true,
+      });
+      const state = {
+        ...defaultInitialState,
+        hostsList: {
+          hosts: [host],
+        },
+      };
+
+      const [StatefulHostsList] = withState(<HostsList />, state);
+
+      renderWithRouter(StatefulHostsList);
+      expect(screen.getByLabelText('Loading')).toBeInTheDocument();
+    });
+
+    it('should request a deregistration when the clean up button in the modal is clicked', async () => {
+      const user = userEvent.setup();
+
+      const host = hostFactory.build({ deregisterable: true });
+      const state = {
+        ...defaultInitialState,
+        hostsList: {
+          hosts: [host],
+        },
+      };
+
+      const [StatefulHostsList, store] = withState(<HostsList />, state);
+
+      renderWithRouter(StatefulHostsList);
+
+      const table = screen.getByRole('table');
+      const cleanUpButton = table.querySelector(
+        'tr:nth-child(1) > td:nth-child(9) > button'
+      );
+
+      await user.click(cleanUpButton);
+
+      expect(
+        screen.getByText(
+          `Clean up data discovered by agent on host ${host.hostname}`
+        )
+      ).toBeInTheDocument();
+
+      const cleanUpModalButton = screen.getAllByRole('button', {
+        name: 'Clean up',
+      })[1];
+
+      await user.click(cleanUpModalButton);
+
+      const actions = store.getActions();
+      const expectedActions = [
+        {
+          type: 'DEREGISTER_HOST',
+          payload: expect.objectContaining({
```

Comment:
I guess this is more of a smell than anything, it's a result of the `data` variable in `HostsList.jsx` not simply sending all data from `hosts`. Is there a reason why we do that?

Reply:
I'm not really sure if I'm following you.
This is a jest option to partially check objects, without the need of comparing all the fields.
In this case, the unique field that I'm interested in the test is the `id`. If I just leave the object itself, the comparison fails, because the passed value has much more fields, that are irrelevant for this usage.
If you want to be purist, i would need to change:
```
        onCleanUp={() => {
          setCleanUpModalOpen(false);
          dispatch(deregisterHost({id: selectedHost.id}));
        }}
```
and only send that value in the payload. But I'm not sure how better this would be honestly, as we are all the time passing the whole host object to the reducers.

Reply:
Yeah I guess there is the purist way above, I just mean to ask why we filter out some of the host data into `data` [here](https://github.com/trento-project/web/pull/1601/files#diff-ece88c9cf0057cb7af611ba3ec1cb6e74bd25f22b05668431918c7fb46fd3379L183)?
(As opposed to simply using the raw `hosts` list, instead)

Reply:
well, the table is not a 1:1 relation with the factory, or the redux state (it doesn't need to be).
That's why there are some divergencies.